internal/service/resource_fetcher.go

```diff
@@ -160,7 +160,7 @@ func (b *ResourceFetcherBuilder) Build() (
 }
 
 // GetResourceTypeID generates a typeID from a search API object.
-func (h *ResourceFetcher) GetResourceTypeID(from data.Object) (resourceTypeID string, err error) {
+func (h *ResourceFetcher) GetResourceTypeID(from data.Object) (resourceTypeId string, err error) {
```

Comment:
changing the name of the argument name to `resourceTypeId` has created a mismatch between that and the name of the function that spells id as `ID` rather than `Id`.   We should be consistent one way or another.   I understand that the JSON reported externally must be `Id`, and to be honest that's my preference for all acronyms in symbols, but the reason the function and other symbols are spelled as `ID` is because of the Google coding guidelines (https://google.github.io/styleguide/go/decisions#initialisms).   I have not seen any statement anywhere that says we are following that but it does seem to be used in some places in this repo (not everywhere).  

IMHO... `Id` is better than `ID`... especially when acronyms are embedded in the middle of a name.  

cc: @bartwensley

Reply:
I would prefer the field names, variable names and function names to all match, but that could be deferred until later as there is probably quite a bit of cleanup to do here.